### PR TITLE
docs/node: Add Signing Key Manager Policy guide

### PR DESCRIPTION
**Pull request by tjanez**

### CODE REVIEW

**Review by kostko — APPROVED** [review]
Nice!

File: docs/node/run-your-node/advanced/signing-key-manager-policy.md [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+
+Make sure you have followed the [Oasis Node binary installation guide] and have
+the Oasis Node binary installed on your system.
+
+### Entity
+
+Similarly to other things, an entity's private key is used to sign a key manager
+policy.
+
+The trusted key manager policy signer set is hard-coded in the key manager's
```
Comment (matevz) [inline docs/node/run-your-node/advanced/signing-key-manager-policy.md]: I don't think the KM set implementation should go under the EXAMPLE admonition (we're using Example admonition frequently in runtime and dApp tutorials). Something more straight forward like this and then remove the EXAMPLE block below?

```suggestion
The trusted key manager policy signer set is hard-coded in the [key manager's source code][oasis-km-signer-set].
```

File: docs/node/run-your-node/advanced/signing-key-manager-policy.md [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+
+:::
+
+
+To provision a new entity, follow the [instructions in our Validator Node
+guide].
+
+:::danger
+
+We strongly advise AGAINST using an entity that is generated with a file-based
```
Comment (matevz) [inline docs/node/run-your-node/advanced/signing-key-manager-policy.md]: We're not using capitalization for emphasizing stuff, but rather bolding the text:

```suggestion
We strongly advise **against using an entity that is generated with a file-based
```

File: docs/node/run-your-node/advanced/signing-key-manager-policy.md [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+signer for singing key manager policies for production key manager, i.e. the
```
Comment (matevz) [inline docs/node/run-your-node/advanced/signing-key-manager-policy.md]:
```suggestion
signer for singing key manager policies for production key manager**, i.e. the
```

**Review by matevz — CHANGES_REQUESTED** [review]
Some minor stylistic changes.

Also, what do you think about moving this guide as a subpage of the *Key Manager Node* chapter similar to how the *Validator Node* chapter contains two subpages (Governance and Amend Commission Schedule).

If not, then at least add the *See Also* section at the end of
https://github.com/oasisprotocol/docs/blob/tjanez/key-manager-policy/docs/node/run-your-node/keymanager-node.md:

```
## See also

<DocCard item={findSidebarItem('/node/run-your-node/advanced/signing-key-manager-policy')} />
```

File: docs/node/run-your-node/advanced/signing-key-manager-policy.md (hunk 1 above)
Comment (tjanez) [reply docs/node/run-your-node/advanced/signing-key-manager-policy.md]: > I don't think the KM set implementation should go under the EXAMPLE admonition (we're using Example admonition frequently in runtime and dApp tutorials).

Ok, agreed. Especially, if EXAMPLE admonitions are used differently in other parts of the docs.

I don't want the docs to imply that the Oasis Key Manager instance is the only one, so I tried to keep that distinction in the updated variant.

File: sidebarNode.js [hunk 4]
```diff
@@ -86,7 +86,17 @@ const sidebars = {
         'node/run-your-node/archive-node',
         'node/run-your-node/paratime-node',
         'node/run-your-node/paratime-client-node',
-        'node/run-your-node/keymanager-node',
+        {
+          type: 'category',
+          label: 'Key Manager Node',
+          link: {
+            type: 'doc',
+            id: 'node/run-your-node/keymanager-node/keymanager-node',
```
Comment (matevz) [inline sidebarNode.js]:
```suggestion
            id: 'node/run-your-node/keymanager-node/README',
```

File: docs/node/run-your-node/advanced/signing-key-manager-policy.md (hunk 2 above)
Comment (tjanez) [reply docs/node/run-your-node/advanced/signing-key-manager-policy.md]: > We're not using capitalization for emphasizing stuff, but rather bolding the text:

Ok, I've changed to bold.

**Review by matevz — CHANGES_REQUESTED** [review]
You'll have to update references from keymanager-node.md to keymanager-node/README.md in the markdown files.

And there is no need to add any backward redirects, because https://docs.oasis.io/node/run-your-node/keymanager-node will remain a valid URL.

File: sidebarNode.js (hunk 4 above)
Comment (tjanez) [reply sidebarNode.js]: Thanks, updated.

File: docs/node/run-your-node/keymanager-node/signing-key-manager-policy.md [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+
+:::danger
+
+We strongly recommend using a dedicated (single-purpose) entity for signing key
+manager policies for production key managers, i.e. the ones deployed on
+Mainnet and connected to a production ParaTime.
+
+:::
+
+
```
Comment (matevz) [inline docs/node/run-your-node/keymanager-node/signing-key-manager-policy.md]:
```suggestion
```

File: docs/node/run-your-node/keymanager-node/signing-key-manager-policy.md [hunk 6]
```diff
@@ -0,0 +1,192 @@
...
+  https://en.wikipedia.org/wiki/Air_gap_\(networking\)
+
+## Define Variables
+
+For easier handling of key manager policy files, define the following variables:
+
+```
+POLICY=path/to/policy.cbor
+KEY=path/to/entity/key.pem
+NAME=myname
```
Comment (matevz) [inline docs/node/run-your-node/keymanager-node/signing-key-manager-policy.md]:
```suggestion
NAME=your_name
```

File: docs/node/run-your-node/keymanager-node/signing-key-manager-policy.md [hunk 7]
```diff
@@ -0,0 +1,192 @@
...
+
+This should output something like the following:
+
+:::tip
+
+This is an example of an actual Oasis Testnet Key Manager policy.
+
+:::
+
+```bash
```
Comment (matevz) [inline docs/node/run-your-node/keymanager-node/signing-key-manager-policy.md]:
```suggestion
```json title="Example of an actual Oasis Testnet Key Manager policy"
```

File: docs/node/run-your-node/keymanager-node/signing-key-manager-policy.md [hunk 8]
```diff
@@ -0,0 +1,192 @@
...
+  }
+}
+```
+
+The `"serial"` key, e.g. `8`, represents the key manager policy's serial number that must
+increase with every update of the key manager policy.
+
+The `"id"` key, e.g.
+`"4000000000000000000000000000000000000000000000004a1a53dff2ae482d"`, represents
+the key manager ParaTime's runtime id.
```
Comment (matevz) [inline docs/node/run-your-node/keymanager-node/signing-key-manager-policy.md]: Either runtime ID or runtime identifier.

```suggestion
the key manager ParaTime's runtime ID.
```

File: docs/node/run-your-node/keymanager-node/signing-key-manager-policy.md [hunk 9]
```diff
@@ -0,0 +1,192 @@
...
+
+The keys below `"enclaves"`, e.g.
+`"ZhD5ufyc/MReZD1qMSKNCRxnkNiZ3BtxqcYdx4+M0N9AJdq369ofvsxONjdgbgISFND0HG0EIv03iyqLiIGEWQ=="`
+and `"jTX8etUcGSQBq3C4WbLlexga7dhQFnwzSJOEmRCPvfRAJdq369ofvsxONjdgbgISFND0HG0EIv03iyqLiIGEWQ=="`,
+represent the identities of the key manager enclaves.
+Each key manager enclave id is comprised of two parts: its `MRENCLAVE` and its
```
Comment (matevz) [inline docs/node/run-your-node/keymanager-node/signing-key-manager-policy.md]:
```suggestion
Each key manager enclave ID is comprised of two parts: its `MRENCLAVE` and its
```

File: docs/node/run-your-node/keymanager-node/signing-key-manager-policy.md [hunk 10]
```diff
@@ -0,0 +1,192 @@
...
+`MRSIGNER`.
+
+Each key manager enclave identity has two lists: `"may_query"` and
+`"may_replicate"`.
+
+Items in `"may_query"` list, e.g.
+`"0000000000000000000000000000000000000000000000000000000000000000"` and
+`"000000000000000000000000000000000000000000000000a6d1e3ebf60dff6c"`,
+represent the runtime ids of the ParaTimes that are allowed to query the key
```
Comment (matevz) [inline docs/node/run-your-node/keymanager-node/signing-key-manager-policy.md]:
```suggestion
represent the runtime IDs of the ParaTimes that are allowed to query the key
```

File: docs/node/run-your-node/keymanager-node/signing-key-manager-policy.md [hunk 11]
```diff
@@ -0,0 +1,192 @@
...
+manager (in this example, the Cipher and the Sapphire ParaTimes running on the
+Testnet).
+
+The items under runtime ids of the ParaTimes, e.g.
+`"c0SidcKhBx3iuonmtXURnFB+qIVkg+nAiaAozAh16ltAJdq369ofvsxONjdgbgISFND0HG0EIv03iyqLiIGEWQ=="` and
+`"LwbLEQ6dv+R5wv5q5CGRZWiEBWGxgCi/gpphcJFQ5zVAJdq369ofvsxONjdgbgISFND0HG0EIv03iyqLiIGEWQ=="`,
+represent the identities of the runtime enclaves.
+Similarly to the key manager enclave id, each runtime enclave id is comprised of
```
Comment (matevz) [inline docs/node/run-your-node/keymanager-node/signing-key-manager-policy.md]:
```suggestion
Similar to the key manager enclave ID each runtime enclave ID is comprised of
```

File: docs/node/run-your-node/keymanager-node/signing-key-manager-policy.md [hunk 12]
```diff
@@ -0,0 +1,192 @@
...
+two parts: its `MRENCLAVE` and its `MRSIGNER`.
+
+Items in `"may_replicate"` list, e.g.
+`"jTX8etUcGSQBq3C4WbLlexga7dhQFnwzSJOEmRCPvfRAJdq369ofvsxONjdgbgISFND0HG0EIv03iyqLiIGEWQ=="`,
+represent the key manager enclave ids to which an existing key manager enclave is
```
Comment (matevz) [inline docs/node/run-your-node/keymanager-node/signing-key-manager-policy.md]:
```suggestion
represent the key manager enclave IDs to which an existing key manager enclave is
```

File: docs/node/run-your-node/keymanager-node/signing-key-manager-policy.md [hunk 13]
```diff
@@ -0,0 +1,192 @@
...
+manager policy files.
+
+:::
+
+## Signing a Policy
+
+Once a key manager policy has been inspected, use the following command to sign
+it:
+
+```
```
Comment (matevz) [inline docs/node/run-your-node/keymanager-node/signing-key-manager-policy.md]:
```suggestion
```bash
```

File: docs/node/run-your-node/keymanager-node/signing-key-manager-policy.md [hunk 14]
```diff
@@ -0,0 +1,192 @@
...
+oasis-node keymanager sign_policy \
+  --keymanager.policy.file $POLICY \
+  --keymanager.policy.key.file $KEY \
+  --keymanager.policy.signature.file $POLICY.$NAME.signed
+```
```
Comment (matevz) [inline docs/node/run-your-node/keymanager-node/signing-key-manager-policy.md:196]: Is there something missing at the end? Should the signature be submitted? Perhaps a sentence in the beginning of the chapter would be nice which describes where does signing of KM policies fit in.

File: docs/node/run-your-node/keymanager-node/signing-key-manager-policy.md [hunk 15]
```diff
@@ -0,0 +1,192 @@
+# Signing Key Manager Policy
+
+This guide will describe how to print and sign an Oasis key manager policy.
```
Comment (matevz) [inline docs/node/run-your-node/keymanager-node/signing-key-manager-policy.md]:
```suggestion
This guide will describe how to print and sign an Oasis [key manager policy].
```

File: docs/node/run-your-node/keymanager-node/signing-key-manager-policy.md [hunk 16]
```diff
@@ -0,0 +1,192 @@
...
+manager policy signer set and hence the key manager itself.
+
+:::
+
+:::info
+
+Currently, Ledger-based signers do not support signing key manager policies.
+
+:::
+
```
Comment (matevz) [inline docs/node/run-your-node/keymanager-node/signing-key-manager-policy.md:72]:
```suggestion

[key manager policy]: https://github.com/oasisprotocol/oasis-core/blob/master/docs/consensus/services/keymanager.md#policies
```

FYI linking to `master` branch above is fine - the cross-repo-link plugin will rewrite it to the local URL.

File: docs/node/run-your-node/keymanager-node/signing-key-manager-policy.md [hunk 17]
```diff
@@ -0,0 +1,192 @@
...
+```
+POLICY=path/to/policy.cbor
+KEY=path/to/entity/key.pem
+NAME=myname
+```
+
+## Printing a Policy
+
+To print and inspect a key manager policy, use the following command:
```
Comment (matevz) [inline docs/node/run-your-node/keymanager-node/signing-key-manager-policy.md:95]:
```suggestion
To print and inspect an existing key manager policy, use the following command:
```

File: docs/node/run-your-node/keymanager-node/signing-key-manager-policy.md [hunk 18]
```diff
@@ -0,0 +1,192 @@
...
+
+```bash
+oasis-node keymanager verify_policy \
+  --keymanager.policy.file $POLICY \
```
Comment (matevz) [inline docs/node/run-your-node/keymanager-node/signing-key-manager-policy.md:99]: Where does the file come from? Can this be dumped from somewhere?

**Review by matevz — APPROVED** [review]
Some suggestions. Otherwise looks good.

File: docs/node/run-your-node/keymanager-node/signing-key-manager-policy.md (hunk 5 above)
Comment (tjanez) [reply docs/node/run-your-node/keymanager-node/signing-key-manager-policy.md]: Fixed

File: docs/node/run-your-node/keymanager-node/signing-key-manager-policy.md (hunk 11 above)
Comment (tjanez) [reply docs/node/run-your-node/keymanager-node/signing-key-manager-policy.md]: I changed the id capitalization. For the other comment I think my previous version sounds a bit better.

File: docs/node/run-your-node/keymanager-node/signing-key-manager-policy.md (hunk 17 above)
Comment (tjanez) [reply docs/node/run-your-node/keymanager-node/signing-key-manager-policy.md:95]: The key manager policy printing is not limited to existing key manager policies.

Usually, one would use this command to print out the contents of an upcoming key manager policy.

File: docs/node/run-your-node/keymanager-node/signing-key-manager-policy.md (hunk 18 above)
Comment (tjanez) [reply docs/node/run-your-node/keymanager-node/signing-key-manager-policy.md:99]: This is out of scope for this guide

File: docs/node/run-your-node/keymanager-node/signing-key-manager-policy.md (hunk 14 above)
Comment (tjanez) [reply docs/node/run-your-node/keymanager-node/signing-key-manager-policy.md:196]: This is out of scope for this guide

**Pull request merged**